TheaterSchedule.BLL/Services/ImageService.cs

```diff
@@ -1,16 +1,125 @@
         {
             string base64Image = Convert.ToBase64String(image);
             return base64Image;
         }
-    }
 
+        public string GeMimeTypeFromImageByteArray(byte[] byteArray)
+        {
+            try
+            {
+                using (MemoryStream stream = new MemoryStream(byteArray))
+                {
+                    using (Image image = Image.FromStream(stream))
+                    {
+                        return ImageCodecInfo.GetImageEncoders().First(codec => codec.FormatID == image.RawFormat.Guid).MimeType;
+                    }
+                }
+            }
+            catch (ArgumentException)
+            {
+                return "text/plain";
+            }
+        }   
+
+        public async Task<ImageBytesDTO> LoadPerformanceMainImageBytesAsync(int id)
+        {
+            ImageBytesDTO imageDTO = null;
+            byte[] image = await TryAddPerformanceImageToCacheAsync(id);
+
+            if (image != null)
+            {
+                await Task.Run(() =>
+                {
+                    imageDTO = new ImageBytesDTO()
+                    {
+                        Image = image,
+                        ImageFormat = GeMimeTypeFromImageByteArray(image)
+                    };
+                });
+            }
+
+            return imageDTO;
+        }
+
+        public async Task<List<ImageBytesDTO>> LoadPerformanceGalleryBytesAsync(int id)
+        {
+            List<ImageBytesDTO> imagesDTO = new List<ImageBytesDTO>();
+            IEnumerable<byte[]> images = await TryAddGalleryImagesToCacheAsync(id);
+
+            await Task.Run(() =>
+            {
+                imagesDTO = images.Select(image => new ImageBytesDTO()
+                {
+                    Image = image,
+                    ImageFormat = GeMimeTypeFromImageByteArray(image)
+                }).ToList<ImageBytesDTO>();               
+            });
+
+            return imagesDTO;
+        }
+
+        public async Task<IEnumerable<byte[]>> TryAddGalleryImagesToCacheAsync(int performanceId)
+        {
+            IEnumerable<byte[]> images = new List<byte[]>();        
+            string key = typeof(Performance).Name + typeof(ImageService).Name + typeof(GalleryImage).Name + performanceId.ToString();
+
+            if (!cache.TryGetValue(key, out images))
+            {
+                images = await imageRepository.GetGalleryImagesAsync(performanceId);
+                if (images.Count() > 0)
+                {
+                    cache.Set(key, images, new MemoryCacheEntryOptions().SetAbsoluteExpiration(TimeSpan.FromMinutes(cacheExpirationMinutes)));
+                }
+            }
+
+            return images;
+        }
+
+        public async Task<byte[]> TryAddPerformanceImageToCacheAsync(int id)
+        {
+            byte[] image = null;
+            string key = typeof(Performance).Name + typeof(ImageService).Name + id.ToString();

```

Comment:
use nameof keyword
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/nameof

Also use string.Format to build key